azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java

```diff
@@ -457,43 +487,171 @@ public DefinitionCreatable withNewAvailabilitySet(String name) {
     }
 
     @Override
-    public DefinitionCreatable withNewAvailabilitySet(AvailabilitySet.DefinitionCreatable creatable) {
-        this.availabilitySetName = creatable.key();
-        this.addCreatableDependency(creatable);
+    public DefinitionCreatable withExistingAvailabilitySet(AvailabilitySet availabilitySet) {
+        this.existingAvailabilitySetToAssociate = availabilitySet;
         return this;
     }
 
+    /**************************************************.
+     * Getters
+     **************************************************/
+
     @Override
-    public DefinitionCreatable withExistingAvailabilitySet(String name) {
-        this.availabilitySetName = name;
-        return this;
+    public String computerName() {
+        return inner().osProfile().computerName();
     }
 
     @Override
-    public VirtualMachine create() throws Exception {
-        if (requiresImplicitStorageAccountCreation()) {
-            withNewStorageAccount(this.key() + UUID.randomUUID().toString());
+    public String size() {
+        return inner().hardwareProfile().vmSize();
+    }
+
+    @Override
+    public OperatingSystemTypes osType() {
+        return inner().storageProfile().osDisk().osType();
+    }
+
+    @Override
+    public String osDiskVhdUri() {
+        return inner().storageProfile().osDisk().vhd().uri();
+    }
+
+    @Override
+    public CachingTypes osDiskCachingType() {
+        return inner().storageProfile().osDisk().caching();
+    }
+
+    @Override
+    public Integer osDiskSize() {
+        return inner().storageProfile().osDisk().diskSizeGB();
+    }
+
+    @Override
+    public List<DataDisk> dataDisks() {
+        return inner().storageProfile().dataDisks();
+    }
+
+    @Override
+    public NetworkInterface primaryNetworkInterface() throws CloudException, IOException {
+        String primaryNicId = primaryNetworkInterfaceId();
+        return this.networkManager.networkInterfaces()
+                .get(ResourceUtils.groupFromResourceId(primaryNicId), ResourceUtils.nameFromResourceId(primaryNicId));
+    }
```

Comment:
This can be done, only issue here is - the cached nic become staled if nic get updated some other way..

Also if we set this during VM initialization time then VirtualMachines().define() will  `throws` right? Setting during refresh is fine as it already marked with `throws`.